flight.py

```diff
@@ -606,31 +632,34 @@ def start_recording():
             except Exception as error:
                 gps = initialize_gt_u7()
             
-            descending_speed_bool =  False
 
-            gps_negative_speed_bool = gps_speed < -2 if gps_speed is not None else False
-            avg_bme_vertical_speed_bool = avg_bme_vertical_speed < -2 if avg_bme_vertical_speed is not None else False
+            if (gps_speed < -2 if gps_speed is not None else False):
+                gps_negative_speed_bool.set_true()
+            if (avg_bme_vertical_speed < -2 if avg_bme_vertical_speed is not None else False):
+                avg_bme_vertical_speed_bool.set_true()
             # cansat should have < 0m/s^2 static acceleration in free fall
-            vertical_acceleration_bool = lowest_z_acceleration < -10
+            if lowest_z_acceleration < -10:
+                vertical_acceleration_bool.set_true()
+            if highest_luminance > avg_highest_luminance + 500 or all_light_sensors_broken:
+                above_avg_luminance_count += 1
 
             bme_works = bme_altitude is not None
             gps_works = gps_altitude is not None
             gyro_works = acceleration_z is not None
 
             working_sensors_count = bme_works + gps_works + gyro_works
-            descending_indicators_count = gps_negative_speed_bool + avg_bme_vertical_speed_bool + vertical_acceleration_bool
+            descending_indicators_count = gps_negative_speed_bool.state + avg_bme_vertical_speed_bool.state + vertical_acceleration_bool.state
             
             # if at least half of the working sensors indicate a descent, descending mode is activated
             if descending_indicators_count >= working_sensors_count - 1:
-                descending_speed_bool = True
+                descending_speed_bool.set_true()
 
             if (
                 (
                     (
                         timestamp - start_ascend_timestamp > 3000 # ascend must be longer than 3s
-                        and descending_speed_bool # half of working sensors indicate a descent
-                        # TODO: do we need this luminance check?
-                        and (highest_luminance > 1000 or all_light_sensors_broken) # the light sensors are outside (> 1000 lux) or dont work
+                        and descending_speed_bool.state # half of working sensors indicate a descent
+                        and above_avg_luminance_count >= 2 # the light sensors are above average luminance for at least 2 iteration or dont work
```

Comment:
ich glaub count above avg luminance ist überflüssig, wir sind eh nur so kurz in ascending, da sollten wir lieber direkt in descending sobald es hell wird